### PR TITLE
0.2.x

### PR DESCRIPTION
**Pull request by sendyhalim**
Add awsUrlGenerator, inject generator to list template variables

### CODE REVIEW

File: lib/awsUrlGenerator.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+var crypto = require('crypto');
+var keystone = require('../');
+var s3Config = keystone.get('s3 config');
+var awsSecretKey = s3Config.secret;
+var awsAccessKey = s3Config.key;
+var awsBucket = s3Config.bucket;
+
+function generateCanonicalString(url, expires) {
+  var strippedBaseUrl = '/' + url.split('/').slice(3).join('/');
+  console.log(strippedBaseUrl);
```
Comment (osumampouw) [inline lib/awsUrlGenerator.js]: is this a debug message?

File: lib/awsUrlGenerator.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  hmac.write(data);
+  hmac.setEncoding('base64');
+  hmac.end();
+
+  return encodeURIComponent(hmac.read());
+}
+
+exports = module.exports = {
+  generate: function(url) {
+    var limit = 3600;
```
Comment (osumampouw) [inline lib/awsUrlGenerator.js]: add comment saying this is the time limit how long token is valid

File: lib/awsUrlGenerator.js (hunk 1 above)
Comment (sendyhalim) [inline lib/awsUrlGenerator.js]: @osumampouw No, I forgot to delete the console.log, I'll do another pull request to fix this :beer:

File: lib/awsUrlGenerator.js (hunk 2 above)
Comment (sendyhalim) [inline lib/awsUrlGenerator.js]: @osumampouw Ok

File: templates/mixins/columns.jade [hunk 3]
```diff
@@ -54,4 +54,7 @@ mixin column_html(value)
 	div.col-value!= value
 
 mixin column_basic(value)
-	div.col-value= value
+	if (value && value.indexOf('amazonaws')!== -1)
```
Comment (osumampouw) [inline templates/mixins/columns.jade]: https://s3-ap-southeast-1.amazonaws.com

coba di test dulu kalo index of jalan ngga pake ini. Soalnya sapa tahu perlu di escape characternya atau apa

**Pull request merged**